musl 1.2.5 exposes ppoll by default, which fails a ppoll exposed test
--
--- a/tests/project/common/103 has header symbol/meson.build	x
+++ a/tests/project/common/103 has header symbol/meson.build	y
@@ -47,10 +47,6 @@
 # on Linux
 if cc.has_function('ppoll') and host_machine.system() == 'linux'
     assert(
-        not cc.has_header_symbol('poll.h', 'ppoll'),
-        'ppoll should not be accessible without _GNU_SOURCE',
-    )
-    assert(
         cc.has_header_symbol('poll.h', 'ppoll', prefix: '#define _GNU_SOURCE'),
         'ppoll should be accessible with _GNU_SOURCE',
     )
